compiler/rustc_middle/src/ty/consts/int.rs

```diff
@@ -1,113 +1,124 @@
+use crate::ty::{self, Ty, TyCtxt};
 use rustc_apfloat::ieee::{Double, Single};
 use rustc_apfloat::Float;
 use rustc_serialize::{Decodable, Decoder, Encodable, Encoder};
 use rustc_target::abi::Size;
+use rustc_type_ir::{IntTy, UintTy};
 use std::convert::{TryFrom, TryInto};
 use std::fmt;
 
-use crate::ty::TyCtxt;
+#[derive(Copy, Clone)]
+enum IntegerTy {
```

Comment:
I checked, but there seem to be no existing enums like this. Surprising...

Reply:
We might want to refactor `Ty` to have a single `Int` variant with a field of this type, rather than 2 top-level `Int` and `Uint` variants.

Reply:
> This is surprising to me. ScalarInt was created to share the Scalar integer handling with const representation. And now we say we don't want that sharing...?

Note that ConstInt isn't used anywhere but printing. Which is probably why there are no uses of the ScalarInt field. As apparent from the diff, all creation sites of ConstInt had to jump through hoops to get a ScalarInt to feed into the new function